logger/handlers/ringbuffer.go

```diff
@@ -0,0 +1,134 @@
+package handlers
+
+import (
+	"container/ring"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/deis/deis/logger/drain"
+	"github.com/deis/deis/logger/syslog"
+)
+
+var projectBuffer = make(map[string]*ring.Ring)
+var ringBufferSize int
+
+func addToBuffer(name string, message string) {
+	currentRing, ok := projectBuffer[name]
+	if !ok {
+		r := ring.New(ringBufferSize)
+		r.Value = message
+		projectBuffer[name] = r
+	} else {
+		r := currentRing.Next()
+		r.Value = message
+		projectBuffer[name] = r
+	}
+}
+
+func RingBufferHandler(bufferSize int, webServicePort int) *Handler {
```

Comment:
Rings are not goroutine-safe, are they? If we have one goroutine reading and another writing, we could end up with some weird race conditions. Have you run this with `go run -race -cpu=2`? Some unit tests would be helpful here.